x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistry.java

```diff
@@ -58,42 +61,49 @@
 public class ModelRegistry {
     public record ModelConfigMap(Map<String, Object> config, Map<String, Object> secrets) {}
 
-    /**
-     * Semi parsed model where inference entity id, task type and service
-     * are known but the settings are not parsed.
-     */
-    public record UnparsedModel(
-        String inferenceEntityId,
-        TaskType taskType,
-        String service,
-        Map<String, Object> settings,
-        Map<String, Object> secrets
-    ) {
-
-        public static UnparsedModel unparsedModelFromMap(ModelConfigMap modelConfigMap) {
-            if (modelConfigMap.config() == null) {
-                throw new ElasticsearchStatusException("Missing config map", RestStatus.BAD_REQUEST);
-            }
-            String inferenceEntityId = ServiceUtils.removeStringOrThrowIfNull(
-                modelConfigMap.config(),
-                ModelConfigurations.INDEX_ONLY_ID_FIELD_NAME
-            );
-            String service = ServiceUtils.removeStringOrThrowIfNull(modelConfigMap.config(), ModelConfigurations.SERVICE);
-            String taskTypeStr = ServiceUtils.removeStringOrThrowIfNull(modelConfigMap.config(), TaskType.NAME);
-            TaskType taskType = TaskType.fromString(taskTypeStr);
-
-            return new UnparsedModel(inferenceEntityId, taskType, service, modelConfigMap.config(), modelConfigMap.secrets());
+    public static UnparsedModel unparsedModelFromMap(ModelConfigMap modelConfigMap) {
+        if (modelConfigMap.config() == null) {
+            throw new ElasticsearchStatusException("Missing config map", RestStatus.BAD_REQUEST);
         }
+        String inferenceEntityId = ServiceUtils.removeStringOrThrowIfNull(
+            modelConfigMap.config(),
+            ModelConfigurations.INDEX_ONLY_ID_FIELD_NAME
+        );
+        String service = ServiceUtils.removeStringOrThrowIfNull(modelConfigMap.config(), ModelConfigurations.SERVICE);
+        String taskTypeStr = ServiceUtils.removeStringOrThrowIfNull(modelConfigMap.config(), TaskType.NAME);
+        TaskType taskType = TaskType.fromString(taskTypeStr);
+
+        return new UnparsedModel(inferenceEntityId, taskType, service, modelConfigMap.config(), modelConfigMap.secrets());
     }
 
     private static final String TASK_TYPE_FIELD = "task_type";
     private static final String MODEL_ID_FIELD = "model_id";
     private static final Logger logger = LogManager.getLogger(ModelRegistry.class);
 
     private final OriginSettingClient client;
+    private Map<String, UnparsedModel> defaultConfigs;
 
     public ModelRegistry(Client client) {
         this.client = new OriginSettingClient(client, ClientHelper.INFERENCE_ORIGIN);
+        this.defaultConfigs = new HashMap<>();
+    }
+
+    public void addDefaultConfigurations(List<UnparsedModel> serviceDefaultConfigs) {
```

Comment:
Generally, I think a method `addDefaultConfiguration` (singular) makes more sense.

Then you don't have the awkward `List.of` in:
```
 registry.addDefaultConfigurations(List.of(oneConfig))
```

For adding a list, you can easily do:
```
allConfig.forEach(registry::addDefaultConfiguration)
```

Reply:
Yes that is better I'm made this change